src/config.js

```diff
@@ -0,0 +1,201 @@
+				if ( !( target[ parts[ i ] ] instanceof Config ) ) {
+					target.set( parts[ i ], new Config() );
+				}
+
+				target = target[ parts[ i ] ];
+			}
+
+			// Values set as pure objects will be treated as Config subsets.
+			if ( utils.isPlainObject( value ) ) {
+				// If the target is an instance of Config (a deep config subset).
+				if ( target[ name ] instanceof Config ) {
+					// Amend the target with the value, instead of replacing it.
+					target[ name ].set( value );
+
+					return;
+				}
+
+				value = new Config( value );
+			}
+
+			// Values will never be undefined.
+			if ( typeof value == 'undefined' ) {
+				value = null;
+			}
+
+			// Call the original set() on the target.
+			Model.prototype.set.call( target, name, value );
+		},
+
+		/**
+		 * Gets the value for a configuration entry.
+		 *
+		 *     config.get( 'name' );
+		 *
+		 * Deep configurations can be retrieved by separating each part with a dot.
+		 *
+		 *     config.get( 'toolbar.collapsed' );
+		 *
+		 * @param {String} name The configuration name.
+		 * @returns {*} The configuration value or `undefined` if the configuration entry was not found.
+		 */
+		get: function( name ) {
+			// The target for this configuration is, for now, this object.
+			//jscs:disable safeContextKeyword
+			var source = this;
+			//jscs:enable
+
+			// The configuration name should be split into parts if it has dots. E.g: `resize.width`.
+			var parts = name.toLowerCase().split( '.' );
+
+			// Take the name of the configuration from the parts. E.g. `resize.width` -> `width`
+			name = parts.pop();
+
+			// Retrieves the source for this configuration recursively.
+			for ( var i = 0; i < parts.length; i++ ) {
+				// The target will always be an instance of Config.
+				if ( !( source[ parts[ i ] ] instanceof Config ) ) {
+					source = null;
+					break;
+				}
+
+				source = source[ parts[ i ] ];
+			}
+
+			// Try to retrieve it from the source object.
+			if ( source && ( typeof source[ name ] != 'undefined' ) ) {
+				return source[ name ];
+			}
+
+			// If not found, take it from the definition.
+			if ( this.definition ) {
+				return this.definition[ name ];
+			}
+
+			return undefined;
+		},
+
+		/**
+		 * Defines the name and default value for configurations. It accepts the same parameters as the
+		 * {@link Config#set set()} method.
+		 *
+		 * On first call, the {@link Config#definition definition} property is created to hold all defined
+		 * configurations.
+		 *
+		 * This method is supposed to be called by plugin developers to setup plugin's configurations. It would be
+		 * rarely used for other needs.
+		 *
+		 * @param {String|Object} nameOrConfigurations The configuration name or an object from which take properties as
+		 * configuration entries.
+		 * @param {*} [value] The configuration value. Used if a name is passed to nameOrConfigurations. If undefined,
+		 * the configuration is set to `null`.
+		 */
+		define: function( name, value ) {
```

Comment:
I'm out of context right now so ignore my message if it misses the point. But I remember that defining config properties was aimed at not only defining defaults, but also what configuration options are out there (possible validation).